File: meok-brand/character-factory/mass_sync.py
import re
import requests
import json
import os

PACK_FILES = [
    '../../meok/ui/src/lib/character-packs/mythological.ts',
    '../../meok/ui/src/lib/character-packs/historical.ts',
    '../../meok/ui/src/lib/character-packs/archetypes.ts',
    '../../meok/ui/src/lib/character-packs/literary.ts'
]

FACTORY_URL = 'http://127.0.0.1:3300/mint'

def sync_packs():
    print("🚀 Starting Mass Character Sync (Success on 201)...")
    
    for file_path in PACK_FILES:
        full_path = os.path.join(os.path.dirname(__file__), file_path)
        if not os.path.exists(full_path):
            continue
            
        print(f"📦 Processing {file_path}...")
        with open(full_path, 'r') as f:
            content = f.read()
            
        blocks = re.split(r'\n\s+([a-z0-9_]+):\s*{', content)
        
        count = 0
        for i in range(1, len(blocks), 2):
            char_id = blocks[i]
            char_body = blocks[i+1]
            
            if char_id in ['dimensions', 'visual', 'evolutionAxes']: continue
            
            name_match = re.search(r"name:\s*['\"](.*?)['\"]", char_body)
            arch_match = re.search(r"archetype:\s*['\"](.*?)['\"]", char_body)
            tags_match = re.search(r"tags:\s*\[(.*?)\]", char_body, re.DOTALL)
            
            if name_match and arch_match:
                name = name_match.group(1)
                arch = arch_match.group(1)
                tags = []
                if tags_match:
                    tags = [t.strip().strip("'").strip('"') for t in tags_match.group(1).split(',') if t.strip()]
                
                payload = {
                    "name": name,
                    "archeType": arch,
                    "ownerId": 'MEOK-FOUNDER',
                    "tools": tags,
                    "mbti": next((t for t in tags if re.match(r'^[IE][SN][TF][JP]$', t)), 'INTJ'),
                    "enneagram": next((t for t in tags if re.match(r'^[1-9]w[1-9]$', t)), '5w6'),
                    "dimensions": { "warmth": 0.5, "energy": 0.5, "whimsy": 0.5, "edge": 0.5, "complexity": 0.5 }
                }
                
                try:
                    r = requests.post(FACTORY_URL, json=payload, timeout=5)
                    if r.status_code in [200, 201]:
                        print(f"  ✨ Minted {name}")
                        count += 1
                    else:
                        print(f"    ❌ Failed {name}: {r.status_code}")
                except Exception as e:
                    print(f"    ❌ Error connecting for {name}: {e}")
                    
        print(f"  ✅ Finished {file_path} ({count} characters)")

if __name__ == "__main__":
    sync_packs()

File: meok-brand/character-factory/mass_sync.js
const fs = require('fs');
const path = require('path');
const axios = require('axios');

const PACK_FILES = [
    '../../meok/ui/src/lib/character-packs/mythological.ts',
    '../../meok/ui/src/lib/character-packs/historical.ts',
    '../../meok/ui/src/lib/character-packs/archetypes.ts',
    '../../meok/ui/src/lib/character-packs/literary.ts'
];

const FACTORY_URL = 'http://localhost:3300/mint';

async function syncPacks() {
    console.log("🚀 Starting Mass Character Sync...");

    for (const file of PACK_FILES) {
        const filePath = path.resolve(__dirname, file);
        if (!fs.existsSync(filePath)) {
            console.warn(`⚠️ Skipping missing file: ${file}`);
            continue;
        }

        console.log(`📦 Processing ${file}...`);
        const content = fs.readFileSync(filePath, 'utf-8');
        
        // Match character ID followed by its object
        const regex = /([a-z0-9_]+): {([\s\S]*?)}[ \t]*,?\n/g;
        let match;
        let count = 0;

        while ((match = regex.exec(content)) !== null) {
            const charId = match[1];
            const charBody = match[2];
            
            if (charId === 'id' || charId === 'dimensions' || charId === 'visual' || charId === 'evolutionAxes') continue;

            try {
                // VERY aggressive cleaning to turn TS object literal into JSON
                let jsonStr = '{' + charBody
                    .replace(/\/\/.*$/gm, '') // Remove comments
                    .replace(/([a-zA-Z0-9_]+):/g, '"$1":') // Quote keys
                    .replace(/'/g, '"') // Single quotes to double
                    .replace(/,(\s*[\]}])/g, '$1') // Trailing commas
                    .replace(/\\"/g, '"') // Unescape double quotes
                    .replace(/\n/g, ' ') + '}';

                const charData = JSON.parse(jsonStr);
                
                if (charData.name) {
                    const payload = {
                        name: charData.name,
                        archeType: charData.archetype,
                        ownerId: 'MEOK-FOUNDER',
                        tools: charData.tags || [],
                        dimensions: charData.dimensions,
                        mbti: charData.tags?.find(t => /^[IE][SN][TF][JP]$/.test(t)) || 'INTJ',
                        enneagram: charData.tags?.find(t => /^[1-9]w[1-9]$/.test(t)) || '5w6'
                    };

                    console.log(`  ✨ Minting ${charData.name}...`);
                    await axios.post(FACTORY_URL, payload);
                    count++;
                }
            } catch (e) {
                // Ignore parse errors for non-character matches
            }
        }
        console.log(`  ✅ Finished ${file} (${count} characters)`);
    }

    console.log("✅ Mass Sync Complete.");
}

syncPacks().catch(console.error);
